src/stateManager/StateManager.ts

```diff
@@ -733,70 +708,158 @@ class StateManager {
-                forkId,
-                transactionCnt
-            ) +
-                this.getTimeoutWaitTimeSeconds()
-        )
+            block.didEveryoneSign(participants) ||
+            blockEntry.onChainTimestamp !== undefined
+        ) {
             return;
-        const response =
+        }
+
+        // Check if participant posted a commitment on-chain
+        const commitmentResponse =
             await this.stateChannelManagerContract.getBlockCallDataCommitment(
                 this.channelId,
                 forkId,
-                transactionCnt,
-                participantAdr
+                blockHeight,
+                participantAddress
+            );
+
+        if (!commitmentResponse.found) {
+            // No commitment posted - proceed with timeout
+            await this.createTimeOutDispute(
+                forkId,
+                blockHeight,
+                participantAddress
+            );
+            return;
+        }
+
+        // Validate the on-chain commitment is legitimate
+        const isValidCommitment = await this.validateBlockCommitment(
+            blockEntry.block,
+            commitmentResponse.blockCalldataCommitment,
+            participantAddress
+        );
+
+        if (!isValidCommitment) {
+            // Invalid commitment - force timeout
+            // TODO
+        }
+    }
+
+    private async validateBlockCommitment(
+        block: Block,
+        blockCalldataCommitment: Hash,
+        participantAddress: Address
+    ): Promise<boolean> {
+        const onChainResult =
+            await this.validationService.fetchBlockCommitmentCalldata(
+                block,
+                blockCalldataCommitment
             );
-        if (response.found) return;
-        //This should be enough since Clock should always lag behind DLT clock
-        const delayTimeSeconds =
-            this.getTimeoutWaitTimeSeconds() -
-            (Clock.getTimeInSeconds() -
-                this.agreementManager.getLatestBlockTimestamp(forkId));
-
-        if (delayTimeSeconds < 0) {
+
+        if (!onChainResult) {
+            return false;
+        }
+
+        const { signedBlock, timestamp } = onChainResult;
+        const expectedCommitment = hash(
+            Codec.encode({ signedBlock, timestamp }, Type.BlockCommitment)
+        );
+
+        const isValid = expectedCommitment === blockCalldataCommitment;
+
+        if (isValid) {
+            // we should have already notified the event listener.
+            // calling the same handler the event lister would have called
+            this.stateChannelEventListener.handleBlockCalldataPosted(
+                this.channelId,
+                blockCalldataCommitment,
+                participantAddress,
+                block.signedBlock,
+                timestamp
+            );
+        }
+
+        return isValid;
+    }
+
+    private async createTimeOutDispute(
```

Comment:
Feel free to leave a `todo` for this (crating a dispute) and a trello card so we don't forget to full this todo, but a trello card exists for creating a dispute that takes into account this and it's just a simple call after it's implemented